test/unit/org/apache/cassandra/db/filter/ColumnFilterTest.java

```diff
@@ -59,6 +68,17 @@ public void columnFilterSerialisationRoundTrip() throws Exception
         testRoundTrip(ColumnFilter.selection(metadata, metadata.partitionColumns().without(v1)), metadata, MessagingService.VERSION_3014);
     }
 
+    @Test
+    public void testToString()
+    {
+        CFMetaData metadata = metadataSupplier.get();
+        ColumnDefinition v1 = metadata.getColumnDefinition(ByteBufferUtil.bytes("v1"));
+
+        Assert.assertEquals("*/*", ColumnFilter.all(metadata).toString());
+        Assert.assertEquals("[v2, v3]", ColumnFilter.selection(metadata.partitionColumns().without(v1)).toString());
+        Assert.assertEquals("*/[v2, v3]", ColumnFilter.selection(metadata, metadata.partitionColumns().without(v1)).toString());
```

Comment:
I think we could easily extend the test coverage of `ColumnFilter#toString` to subselections if, for example, we declare one of the columns in the table metadata as a not-frozen collection:
```java
.addRegularColumn("v3", SetType.getInstance(Int32Type.instance, true))
```
and then we test:
```java
Assert.assertEquals("[]", ColumnFilter.selectionBuilder().build().toString());
Assert.assertEquals("[v3[1], v3[2]]", ColumnFilter.selectionBuilder().select(v3, c2).select(v3, c1).build().toString());
Assert.assertEquals("[v3[1:2]]", ColumnFilter.selectionBuilder().slice(v3, c1, c2).build().toString());
Assert.assertEquals("*/[v3[1], v3[2]]", ColumnFilter.allColumnsBuilder(metadata).select(v3, c2).select(v3, c1).build().toString());
Assert.assertEquals("*/[v3[1:2]]", ColumnFilter.allColumnsBuilder(metadata).slice(v3, c1, c2).build().toString());
```

Reply:
ok, will add that, thank you